crates/libs/wallet/src/error_handling.rs

```diff
@@ -0,0 +1,15 @@
+use tonic::Status;
+
+pub enum ErrorClass {
+    InvalidProof,
+    AlreadySpent,
+    Recoverable,
+}
+
+pub fn classify_grpc_error(err: &Status) -> ErrorClass {
+    match err.code() {
+        tonic::Code::InvalidArgument => ErrorClass::InvalidProof,
+        tonic::Code::AlreadyExists => ErrorClass::AlreadySpent,
```

Comment:
I don't think this will work.
A Swap can fail because the outputs are invalid and will return InvalidArgument, but the proof will be alright actually.

The node have to be the one doing the classification and giving the wallet the detailed info about what proof is invalid.
You will have to parse the actual error message to get the info out of it.